polar-core/src/bindings.rs

```diff
@@ -153,6 +153,21 @@ impl BindingManager {
         Self::default()
     }
 
+    fn partial_bind(&mut self, partial: Operation, var: &Symbol, val: Term) -> PolarResult<Goal> {
+        match partial.ground(var, val.clone()) {
+            None => Err(RuntimeError::IncompatibleBindings {
```

Comment:
I'm not sure we should be using a runtime error here to mean "the grounding failed". Runtime errors are supposed to capture the VM entering invalid states, or similar. Would it be possible to instead return a backtrack goal?

Reply:
You're totally right, this is just pre-existing logic I didn't want to mess with. You can see in `vm.rs` most callers to `bind` check for an error and backtrack if one happens.